target_chains/ethereum/entropy_sdk/solidity/MockEntropy.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+interface IEntropyConsumer {
+    function _entropyCallback(
+        uint64 sequenceNumber,
+        address provider,
+        bytes32 randomNumber
+    ) external;
+}
+
+contract MockEntropy {
+    uint64 public sequenceNumber;
+    uint128 public constant FEE = 0.000015 ether;
+
+    function getDefaultProvider() external view returns (address) {
+        return address(this);
+    }
+
+    function getFee(address provider) external pure returns (uint128) {
+        require(provider != address(0), "Invalid provider address");
```

Comment:
provider should be the same as `getDefaultProvider()`

Reply:
Revised to be same as getDefaultProvider()